scalding-core/src/main/scala/com/twitter/scalding/ReferencedClassFinder.scala

```
@@ -34,22 +37,29 @@ object ReferencedClassFinder {
    * referred to in a field
    */
   def findReferencedClasses(outerClass: Class[_]): Set[Class[_]] = {
-    val scalaPackage = Package.getPackage("scala")
-    val mirror = universe.runtimeMirror(outerClass.getClassLoader)
-    val scalaType = mirror.classSymbol(outerClass).toType
-    (for {
-      field <- outerClass.getDeclaredFields
-      if baseContainers.exists(_.isAssignableFrom(field.getType))
-      scalaSignature = scalaType.member(universe.stringToTermName(field.getName)).typeSignature
-      clazz <- getClassesForType(mirror, scalaSignature)
-      /* The scala root package contains a lot of shady stuff, eg compile-time wrappers (scala.Int/Array etc),
+    try {
```

I think it's fine to catch the errors and fall back to empty Set, but we should make the try block as specific as possible. It seems that it could wrap only `mirror.classSymbol` and catch only these cases:

1. RuntimeException with message containing the string "error reading Scala signature", adding an explanation that it happens when the class is not in the classpath.
2. scala.reflect.internal.Symbols$CyclicReference, adding a link to the scala bug

---

Yeah I don't mind a specific try block and only catching those 2 exceptions. We'd work around the 2 issues we've seen so far. Risk is if there are others, we'll find out with job failures rather than an error in the logs. Given that this feature / function is a nice-have, I'm thinking it's not too bad to catch the way we're currently doing.